### ISSUE TITLE
Error exporting traces to Lightstep

### ISSUE THREAD

**Issue by DanTulovsky** [emissary-ingress/emissary#3275]
**Describe the bug**
I am having trouble getting Ambassador to export data to Lightstep.

This is my ambassador tracing config:

```
---
apiVersion: getambassador.io/v2
kind: TracingService
metadata:
  name: tracing
  namespace: ambassador
spec:
  service: "ingest.lightstep.com:443"
  # config: {}
  driver: lightstep
  config: { access_token_file: /etc/lightstep-access-token/token.txt }
  tag_headers:
    - ":authority"
    - ":path"
  sampling:
    overall: 100
```

I can confirm that the token file is properly mounted and has the correct token (it's used in other places as well).

Perhaps the issue is that `:scheme` is `http` vs `https`?  This page (https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md) seems to imply `https` exists.

**Versions (please complete the following information):**
 - Ambassador version: 1.11.2
 - Kubernetes environment: GKE
 - Version: 1.19.7-gke.1500

**Additional context**

Here's a debug log entry of, what I believe, is ambassador trying to send traces:

```
[2021-03-15 16:28:56.921][46][debug][router] [source/common/router/router.cc:426] [C0][S9630094733567160337] cluster 'cluster_tracing_ingest_lightstep_com_443_ambassador' match for URL '/lightstep.collector.CollectorService/Report'
[2021-03-15 16:28:56.923][46][debug][router] [source/common/router/router.cc:583] [C0][S9630094733567160337] router decoding headers:
':method', 'POST'
':path', '/lightstep.collector.CollectorService/Report'
':authority', 'cluster_tracing_ingest_lightstep_com_443_ambassador'
':scheme', 'http'
'te', 'trailers'
'grpc-timeout', '5000m'
'content-type', 'application/grpc'
'x-envoy-internal', 'true'
'x-forwarded-for', '10.0.6.8'
'x-envoy-expected-rq-timeout-ms', '5000'

[2021-03-15 16:28:56.923][46][debug][pool] [source/common/http/conn_pool_base.cc:71] queueing request due to no available connections
[2021-03-15 16:28:56.923][46][debug][pool] [source/common/conn_pool/conn_pool_base.cc:53] creating a new connection
[2021-03-15 16:28:56.923][46][debug][client] [source/common/http/codec_client.cc:35] [C2386] connecting
[2021-03-15 16:28:56.923][46][debug][connection] [source/common/network/connection_impl.cc:753] [C2386] connecting to 35.222.202.121:443
[2021-03-15 16:28:56.923][46][debug][connection] [source/common/network/connection_impl.cc:769] [C2386] connection in progress
[2021-03-15 16:28:56.923][46][debug][http2] [source/common/http/http2/codec_impl.cc:1063] [C2386] updating connection-level initial window size to 268435456
[2021-03-15 16:28:56.925][46][debug][connection] [source/common/network/connection_impl.cc:616] [C2386] connected
[2021-03-15 16:28:56.925][46][debug][client] [source/common/http/codec_client.cc:73] [C2386] connected
[2021-03-15 16:28:56.925][46][debug][pool] [source/common/conn_pool/conn_pool_base.cc:146] [C2386] attaching to next request
[2021-03-15 16:28:56.925][46][debug][pool] [source/common/conn_pool/conn_pool_base.cc:73] [C2386] creating stream
[2021-03-15 16:28:56.925][46][debug][router] [source/common/router/upstream_request.cc:342] [C0][S9630094733567160337] pool ready
[2021-03-15 16:28:56.926][46][debug][misc] [source/common/network/io_socket_error_impl.cc:30] Unknown error code 104 details Connection reset by peer
[2021-03-15 16:28:56.926][46][debug][connection] [source/common/network/connection_impl.cc:584] [C2386] remote close
[2021-03-15 16:28:56.926][46][debug][connection] [source/common/network/connection_impl.cc:208] [C2386] closing socket: 0
[2021-03-15 16:28:56.926][46][debug][client] [source/common/http/codec_client.cc:92] [C2386] disconnect. resetting 1 pending requests
[2021-03-15 16:28:56.926][46][debug][client] [source/common/http/codec_client.cc:115] [C2386] request reset
[2021-03-15 16:28:56.926][46][debug][pool] [source/common/conn_pool/conn_pool_base.cc:98] [C2386] destroying stream: 0 remaining
[2021-03-15 16:28:56.926][46][debug][router] [source/common/router/router.cc:1022] [C0][S9630094733567160337] upstream reset: reset reason connection termination
[2021-03-15 16:28:56.926][46][debug][http] [source/common/http/async_client_impl.cc:99] async http request response headers (end_stream=true):
':status', '200'
'content-type', 'application/grpc'
'grpc-status', '14'
'grpc-message', 'upstream connect error or disconnect/reset before headers. reset reason: connection termination'
```

**Comment by illinar** [comment]
Ran into the same situation. In our case it seems to be related to the firewall. We had L7 FQDN rule for the `ingest.lightstep.com`, but the connection is actually made using IP, as evident from the log `connecting to 35.222.202.121:443`. After we added IP-based rule the connection termination errors disappear.

Although we are still unable to make tracing work, since we are seeing 400 status codes now.
